### PR TITLE
Amorale/lp 41 ccache support

### PR DESCRIPTION
**Pull request by AlessioMorale**
It only misses Windows supports.
I'm going to add as a separate Issue/PR as I'm currently working at an easy way to compile it

### CODE REVIEW

**Comment by AlessioMorale** [comment]
fixed

**Comment by paul-jewell** [comment]
Compilation on gentoo linux (core i7). following ccache -C and make clean:
time make -j4 gcs:
real    6m56.550s
user    23m35.890s
sys     1m52.447s
> make clean && time make -j4 gcs:
real    3m12.250s
user    8m50.165s
sys     0m51.974s
No issues with the compilation. Now testing the flight modules compilation.

**Comment by AlessioMorale** [comment]
To enable the use of ccache add the following option into the config file in the project root:
```
export CCACHE:=ccache
```

File: make/tools.mk [hunk 1]
```diff
@@ -204,7 +204,7 @@ endif
 # Command to extract version info data from the repository and source tree
 export VERSION_INFO = $(PYTHON) $(ROOT_DIR)/make/scripts/version-info.py --path=$(ROOT_DIR)
 
-##############################
+##############################s
```
Comment (parched) [inline make/tools.mk]: s?

Comment (AlessioMorale) [inline make/tools.mk]: amended while adding the export within tools.mk

File: make/tools.mk [hunk 2]
```diff
@@ -204,6 +204,8 @@ endif
 # Command to extract version info data from the repository and source tree
 export VERSION_INFO = $(PYTHON) $(ROOT_DIR)/make/scripts/version-info.py --path=$(ROOT_DIR)
 
+export CCACHE:=$(CCACHE)
+
```
Comment (parched) [inline make/tools.mk]: Can't this be just 
`export CCACHE`

Comment (AlessioMorale) [inline make/tools.mk]: facepalm :) 
fixed

**Comment by parched** [comment]
I am happy for this to go in now. I think you should squash the two commits before merging though, no point having partial reversion in a branch before it goes into next.

**Comment by AlessioMorale** [comment]
Fixed, merging
Thanks for your help!

**Pull request merged**